### DISCUSSION TITLE
Docker and API map to 0.0.0.0

### DISCUSSION THREAD

**Discussion by Mario-o-o-o** [Tribler/tribler#8335] [Ideas]
~~Docker and API map to 0.0.0.0~~
Docker and **webUI** map to 0.0.0.0

Hi

[On the forum](https://forum.tribler.org/t/tribler-in-docker-network-mode-and-iptables/7371), I explained that I wanted to use Tribler via docker, with strict iptables rules (default policy DROP).  
  
In `network_mode: "host"`, Tribler and/or libtorrent attach a port to each IP ported by the host, and with X containers, I have X ports listening.  
It's a mess. If I'm not mistaken, these ports (3XXXX) are linked to ‘Local Peer Discovery’.  
So I want Tribler to have its own docker network to isolate it.  
However, I obviously can't map access to the web UI via ...
```
    ports:
      - "127.0.0.1:8085:127.0.0.1:8085"
```
Is it possible to have an option to create a web UI that listens on 0.0.0.0 rather than 127.0.0.1 in the container, please?  
  
To continue with my post in the forum, are the only ports to be authorised in iptables on the host really ...
- 8090 and 8091 # ipv8
- from 6881 to 6889 # libtorrent
in tcp and udp please?  
  
Thank you in advance for your help

**Comment by qstokkink** [comment]
These settings should be mostly configurable through a configuration file.

First, you should locate your "state directory". Depending on how you run your Docker container, your state directory will end up in a different folder. For example, if you follow https://tribler.readthedocs.io/en/latest/basics/docker.html#running, your state directory will end up in `~/.Tribler`.

In the state directory, you should find the `configuration.json`. Right now, we only have the bleeding edge Docker package, so this should appear in the `git` version directory inside your state directory.

If you're looking to change your API host (TCP) to `0.0.0.0`, look for the following entries and change them accordingly:

```json
{
    "api": {
         ...
        "http_port": 8085,
        "http_host": "127.0.0.1",
         ...
    }
    ...
}
```

You can do the same thing with the `ip` and `port` under `"ipv8"` `"interfaces"` and disable IPv6 support by removing the entire definition from the list of `"interfaces"`. This should be open for UDP.

For libtorrent, you can only change the `"libtorrent"` `"port"` field. I believe this should be TCP/UDP.

I think the above should work, but this is not something I play with daily. Please let me know if any of this doesn't work out.

**Comment by Mario-o-o-o** [comment]
@qstokkink Ooooh, it's as simple as that, it's perfect!  
I forced the private IP address into the tribler docker container, changed "~/.Tribler/git/configuration.json" from `"http_host": "127.0.0.1",` to `"http_host": "private_ip",` then restarted the container.
Thanks a lot!
It is obviously not possible to map the webUI to 0.0.0.0, the container does not restart with the error "_xvfb-run: error: Xvfb failed to start_".

I don't necessarily need to force the modification of tribler and/or libtorrent ports at the moment, just to make sure which ports to open and with which protocol (tcp, udp).

Have a nice day
